### DISCUSSION TITLE
openModal function hides any existing modals

### DISCUSSION THREAD

**Discussion by atumas-bananamilk** [mantinedev/mantine#4945] [Q&A]
### What package has an issue

@mantine/core

### Describe the bug

Why is "openModal" function hiding / replacing all previously opened modals?

I want to see a modal, then click another button to open a modal of **different size** on top of it (so I see previously opened modal in the back), and it's not the same as highlighted in here: https://mantine.dev/others/modals/#multiple-opened-modals which just replaces / hides the previously opened modal.

This is quite basic functionality, many component libraries support it.

Why doesn't something like this exist?

### What version of @mantine/* packages do you have in package.json? (Note that all @mantine/* packages must have the same version in order to work correctly)

^6.0.0

### Do you know how to fix the issue

None

### Are you willing to participate in fixing this issue and create a pull request with the fix

None

**Comment by rtivital** [comment]
> Why is "openModal" function hiding / replacing all previously opened modals?

It is the implementation of modals manager. You can stack as many modals as needed with Modal component. 

> This is quite basic functionality, many component libraries support it.

What other libraries provide modals manager?

**Reply by atumas-bananamilk** [reply]
"It is the implementation of modals manager. You can stack as many modals as needed with Modal component."
Yes so how do we stack those modals by using functions (openModal), any example? Modal management on our own is not ideal, component library should help us out. openModal function is already able to open another modal, so why not have a flag on this function?

"What other libraries provide modals manager?"
It's not about "providing a modals manager". What's the point of the modals manager if it doesnt help in cases when there's multiple modals? Obviously orchestrating (showing / hiding) modals on our own is not ideal, you know that.

Other libraries supporting modals manager:
- Saasui: https://saas-ui.dev/docs/components/overlay/modals-manager

**Reply by rtivital** [reply]
Okay, I've got your feature request. I do not have the capacity to work on the modals package any time soon, you are welcome to submit a PR with non-breaking changes.

**Reply by atumas-bananamilk** [reply]
Thanks for understanding.
